src/main/java/com/exasol/adapter/dynamodb/dynamodbmetadata/DynamodbTableMetadataFactory.java

```diff
@@ -0,0 +1,76 @@
+package com.exasol.adapter.dynamodb.dynamodbmetadata;
+
+import java.util.List;
+import java.util.stream.Collectors;
+
+import com.amazonaws.services.dynamodbv2.AmazonDynamoDB;
+import com.amazonaws.services.dynamodbv2.model.KeySchemaElement;
+import com.amazonaws.services.dynamodbv2.model.TableDescription;
+import com.exasol.adapter.dynamodb.documentpath.DocumentPathExpression;
+
+/**
+ * This class build {@link DynamodbTableMetadata} by fetching the required information using a {@code describeTable}
+ * call to DynamoDB.
+ */
+public class DynamodbTableMetadataFactory {
+
+    /**
+     * Builds {@link DynamodbTableMetadata} for a DynamoDB table.
+     * 
+     * @param connection DynamoDB connection used for {@code describeTable} call
+     * @param tableName  DynamoDB table name
+     * @return {@link DynamodbTableMetadata} describing the keys and indexes of the table
+     */
+    public DynamodbTableMetadata forDynamodbTable(final AmazonDynamoDB connection, final String tableName) {
+        final TableDescription tableDescription = connection.describeTable(tableName).getTable();
+        final DynamodbKey primaryKey = extractKey(tableDescription.getKeySchema());
+        final List<DynamodbKey> localIndexes = extractLocalSecondaryIndex(tableDescription);
+        final List<DynamodbKey> globalIndexes = extractGlobalSecondaryIndex(tableDescription);
+        return new DynamodbTableMetadata(primaryKey, localIndexes, globalIndexes);
+    }
+
+    private DynamodbKey extractKey(final List<KeySchemaElement> keySchema) {
+        final DocumentPathExpression partitionKey = extractPartitionKey(keySchema);
+        final DocumentPathExpression sortKey = extractSortKey(keySchema);
+        return new DynamodbKey(partitionKey, sortKey);
+    }
+
+    private DocumentPathExpression extractPartitionKey(final List<KeySchemaElement> keySchema) {
+        for (final KeySchemaElement keySchemaElement : keySchema) {
+            if (keySchemaElement.getKeyType().equals("HASH")) {
+                final String keyName = keySchemaElement.getAttributeName();
+                return new DocumentPathExpression.Builder().addObjectLookup(keyName).build();
+            }
+        }
+        throw new IllegalStateException("Could not find partition key. "
+                + "That's strange because each Dynamodb table must define a partition key.");
```

Comment:
I don't like this "That's strange" in the exception message :D

Reply:
better like this?